qutip/tests/test_random.py

```diff
@@ -58,26 +58,48 @@ def test_rand_dm_ginibre_rank():
     rho = rand_dm_ginibre(5, rank=3)
 
     rank = sum([abs(E) >= 1e-10 for E in rho.eigenenergies()])
-    assert_(rank == 3)
+    assert rank == 3
 
 def test_rand_super_bcsz_cptp():
     """
     Random Qobjs: Tests that BCSZ-random superoperators are CPTP.
     """
     S = rand_super_bcsz(5)
-    assert_(S.iscptp)
+    assert S.iscptp
 
 def check_func_dims(func, args, kwargs, dims):
     # TODO: promote this out of test_random, as it's generically useful
     #       in writing tests.
     resdims = func(*args, **kwargs).dims
-    assert_(resdims == dims, "Checking {}; epected dimensions of {}, got {}.".format(func.__name__, dims, resdims))
+    assert resdims == dims
+
+def check_func_N(func, args, kwargs,expected_shape):
+    # Here, expected_shape is supposed to be of form N X 1. When assigning
+    # this to a tuple, value 1 cannot be used.
+    number_of_rows , column_number = expected_shape
```

Comment:
I'm not quite sure what you meant by the second sentence in the comment.  Also, what's the reason for `number_of_rows` and `column_number`?  They're not used at all (and it's kind of odd that you called one `number_of_<x>` and the other `<x>_ number`!).

Reply:
Maybe I ought to have chosen different variable names. The issue is that if I use `N,1 == expected_shape`, I end up with 
`SyntaxError : Can't assign to a literal`. I cannot use tuple `(N,1)` in function definition directly. So, I tried to use the method defined [here](https://stackoverflow.com/a/20035625/10241324). 

By default, `column_number == 1` whereas `number_of_rows` is determined by N or dims.

Reply:
But I'm not sure why you want to assign anything at all?  What are you trying to achieve?

Reply:
It's to do with how `expected_shape` is input in `check_func_N`. 

What I have done is for `expected_shape` to be input as a tuple - (`number_of_rows`,`column_number`). Here, I can input `expected_shape==(N,1)` in `check_func_N` and then at the step where both shapes are checked `new_state_shape==expected_shape`, `(N,1)==(N,1)`. 

If I do not do this extra step then the step fails due to `(N,1)==((N,1),1)`. In addition, if I don't unpack the tuple using  (`number_of_rows`,`column_number`) then the function won't accept `expected_shape==(N,1)`. 

I could change the input for `expected_shape` to be just N instead of `(N,1)`. This could help avoid above step.

Reply:
Huh?  I really don't understand what you're saying at all.  The unpacking has _no_ effect on what `expected_shape` is allowed to be, and no effect on the rest of the code.  This would work completely fine:
```python
def check_function_shape(func, args, kwargs, expected_shape):
    test_shape = func(*args, **kwargs).shape
    assert test_shape == expected_shape

check_function_shape(rand_ket, (6,), {}, (6, 1))
```

Reply:
Yeah, what you have does work. Unfortunately, I cannot replicate the issue that was causing a failure `(N,1)==((N,1),1)`. 
I should spend time to document all errors properly 😅

The new changes accepts `N` and checks the shape as `(N,1)`.

Reply:
Ok, it's good that you understand what went wrong.  There's no need to have both lines of
```python
assert test_shape == (N, 1)
assert test_shape[0] == N
```
because the second line can _never_ fail if the first line passes (and `test_shape` is a tuple), but that's ok.